lib/config/config.js

```diff
@@ -523,6 +532,25 @@ function getLoggingPreambleData (config, sources, configFilePath) {
   return result
 }
 
+/**
+ * Takes an URL string and redacts user & password info if present in the basic
+ * auth part
+ *
+ * @param {String} urlStr The URL to strip sensitive info
+ * @returns {String} url with basic auth REDACTED
+ */
+function sanitizeUrl (urlStr) {
+  const url = parseUrl(urlStr)
+  let result = url.href
+
+  if (url.username || url.password) {
+    result = result.replace(url.username, REDACTED)
+    result = result.replace(url.password, REDACTED)
+  }
+
+  return result
+}
```

Comment:
I think this may not be correct.

Doing:

```
% ELASTIC_APM_DISABLE_SEND=true ELASTIC_APM_SERVER_URL='http://user@example.com/file.html' node examples/trace-http.js | ecslog
...
    config: {
        "serviceName": {
            "source": "default",
            "value": "elastic-apm-node",
            "commonName": "service_name"
        },
        "serviceVersion": {
            "source": "default",
            "value": "3.46.0",
            "commonName": "service_version"
        },
        "serverUrl": {
            "source": "environment",
            "value": "[REDACTED]http://[REDACTED]@example.com/file.html",
            "commonName": "server_url"
        },
```